ActionSheet.android.js

```diff
@@ -110,26 +124,74 @@ class ActionGroup extends React.Component {
           onPress={() => onSelect(i)}
           style={styles.button}>
           {iconElement}
-          <Text style={[styles.text, { color }, textStyle]}>{options[i]}</Text>
+          <Text style={[styles.text, textStyle, { color }]}>{options[i]}</Text>
         </TouchableNativeFeedbackSafe>
       );
 
-      if (i < startIndex + length - 1) {
-        optionViews.push(<View key={`separator-${i}`} style={styles.rowSeparator} />);
+      if (showSeparators && i < startIndex + length - 1) {
+        optionViews.push(this._renderRowSeparator(i));
       }
     }
 
     return (
       <View style={styles.groupContainer}>
+        {this._renderTitleContent()}
         <ScrollView>{optionViews}</ScrollView>
       </View>
     );
   }
+
+  _renderRowSeparator(key) {
+    return <View key={key ? `separator-${key}` : null} style={styles.rowSeparator} />;
+  }
+
+  _renderTitleContent()  {
+    const { title, titleTextStyle, message, messageTextStyle, showSeparators } = this.props;
+
+    if (!title && !message) {
+      return null;
+    }
+
+    return (
+      <View>
+        <View style={[styles.titleContainer, { paddingBottom: showSeparators ? 24 : 16 }]}>
+          {title ? <Text style={[styles.title, titleTextStyle]}>{title}</Text> : null}
```

Comment:
you can do `title && <Text /> ...` here instead of a ternary

Reply:
@peterpme `title && <Text />` doesn't work in the infrequent case that `title === 0`, in which case it produces an error that it needs to be contained in a `<Text>` element, for which reason I always use ternary in react.

Reply:
@JulianKingman 😁 yup that’s why we always do ternary as well. We’ve been burnt by not explicitly returning null in the past where I work.

Reply:
I usually write `{!!title && <Text />}` which is shorter

Reply:
Need to know which way is preferred...

Reply:
@jstheoriginal use @maxaggedon method, that should be fine or cast it to a boolean `Boolean(title) && <Text />`